resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -41,28 +41,37 @@ func (e storageAccountEnricher) Enrich(ctx context.Context, cycleMetadata cycle.
 		return fmt.Errorf("storageAccountEnricher: error while getting subscription: %w", err)
 	}
 
-	diagSettings, err := e.provider.ListDiagnosticSettingsAssetTypes(ctx, cycleMetadata, lo.Keys(subscriptions))
-	if err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting diagnostic settings: %w", err)
+	var errs []error
+
+	if err := e.addUsedForActivityLogsFlag(ctx, cycleMetadata, assets, lo.Keys(subscriptions)); err != nil {
+		errs = append(errs, err)
 	}
-	e.addUsedForActivityLogsFlag(assets, diagSettings)
 
 	storageAccounts := lo.Filter(assets, func(item inventory.AzureAsset, index int) bool {
 		return item.Type == inventory.StorageAccountAssetType
 	})
 
 	if err := e.addStorageAccountBlobServices(ctx, storageAccounts, assets); err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting data protection settings: %w", err)
+		errs = append(errs, fmt.Errorf("storageAccountEnricher: error while getting data protection settings: %w", err))
 	}
 
 	if err := e.addStorageAccountServicesDiagnosticSettings(ctx, storageAccounts, assets); err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting services diagnostic settings: %w", err)
+		errs = append(errs, fmt.Errorf("storageAccountEnricher: error while getting services diagnostic settings: %w", err))
+	}
+
+	if len(errs) > 0 {
+		return errors.Join(errs...)
```

Comment:
_[Mostly wondering]_

Nothing wrong with this approach, I just wonder why do this way and not directly `errs = errors.Join(errs, err)` on each error?

Reply:
Actually, `errors.Join` does not append to an existing `joinError` if this is the first argument or something. 
It creates a new `joinError` and filters out the `nil`s. 

So basically, the [implementation](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/errors/join.go;l=15) logic of `errors.Join` is to put multiple operations under the function, and it will filter out the nils and return an error **if at least one** error appeared.

e.g. 
```golang
err := errors.Join(
    op1(),
    op2(),
    op3(),
)

// handle if err != nil
```

by doing something like `errAgg = errors.Join(errAgg, err)` what will be created is something like this:

```golang
&joinError{
	errs: []error{
		&joinError{
			errs: []error{
				&joinError{
					errs: []error{
						//error a
					},
				},
				//error b
			},
		},
		//error c
	},
	//error d
}
```
Which will work "as expected" (in terms of `.Error()` and `errors.{Is,As}()` handling (because of recursive logic) but seemed "cleaner" to me to create just a single slice.

Reply:
It seems `.Is`,`.As` and `.Error` would all work because it's recursive in case of join [.Is impl](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/errors/wrap.go;l=63;bpv=0), [.As impl](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/errors/wrap.go;l=125;bpv=0)
and joinError [.Error impl](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/errors/join.go;l=46;bpv=0)

Reply:
Ah I misread. I do think the code is cleaner the other way. This way the inner structure is cleaner 🤷 

No deal breaker, but I'm still in favor of the  errAgg = errors.Join(errAgg, err) :P